File: EduMS.Backend/src/EduMS.Domain/Entities/M1_SchoolAdmin/GradeCapacity.cs
using EduMS.Domain.Common;

namespace EduMS.Domain.Entities;

/// <summary>
/// الصفوف الدراسية والطاقة الاستيعابية - Grade-level capacity configuration extracted from ZIP ERD GradeCapacity table (lines 282-302).
/// Manages maximum students per class, section count, and enrollment tracking per grade.
/// </summary>
public class GradeCapacity : BaseAuditableEntity
{
    public long SchoolAcademicYearId { get; set; }
    public long SchoolLevelId { get; set; }
    public string GradeLevelCode { get; set; } = string.Empty;  // "1", "10" (Grade 1 Primary, Grade 1 Secondary)
    public string GradeNameAr { get; set; } = string.Empty;     // الأول الابتدائي
    public string? GradeNameEn { get; set; }
    public int MaxStudentsPerSection { get; set; }
    public int MaxSectionsCount { get; set; }
    public int CurrentEnrolledCount { get; set; }
    public int GenderAllocation { get; set; } = 1; // 1=Boys, 2=Girls, 3=Mixed
    public bool IsActive { get; set; } = true;
    public string? Notes { get; set; }

    // Navigation Properties
    public virtual SchoolAcademicYear? AcademicYear { get; set; }
    public virtual SchoolLevel? SchoolLevel { get; set; }
}

File: 04_Frontend_Application/Edums-Frontend/src/app/modules/m1-school-office/interface/grade-capacity.ts
import type { GenderAllocation } from './common';

export interface GradeCapacity {
  id: number;
  schoolAcademicYearId: number;
  schoolLevelId: number;
  gradeLevelCode: string;
  gradeNameAr: string;
  gradeNameEn: string | null;
  maxStudentsPerSection: number;
  maxSectionsCount: number;
  currentEnrolledCount: number;
  genderAllocation: GenderAllocation;
  isActive: boolean;
  notes: string | null;
}

export type CreateGradeCapacityDto = Omit<GradeCapacity, 'id' | 'isActive'>;

export type UpdateGradeCapacityDto = Omit<GradeCapacity, 'isActive'>;
